src/server/index.js

```diff
@@ -41,14 +41,30 @@ class KiteServer extends Emitter {
     this.id = createId()
     this.server = null
 
-    this.api = new KiteApi({
-      auth: this.options.auth,
-      methods: this.options.api,
-    })
+    this.setApi(
+      new KiteApi({
+        auth: this.options.auth,
+        methods: this.options.api,
+      })
+    )
 
     this.currentToken = null
   }
 
+  setApi(api) {
+    if (api instanceof KiteApi) {
+      this.api = api
+    } else {
+      throw new Error('A valid KiteApi instance is required!')
+    }
+
+    if (this.server != null) {
+      this.logger.info(`API change, restarting running server...`)
+      this.close()
+      this.listen(this.port)
```

Comment:
I think we should listen for `close` event here before calling `listen` again. And maybe we can abstract these operations into a `reconnect` method?

Reply:
`reconnect` would be better.